src/components/schedule-visualizer/index.tsx

```diff
@@ -215,23 +220,45 @@ export default function ScheduleVisualizer(props: ScheduleVisualizerProps): Reac
     setCurLevelName(e.name);
   }
 
-  function getTrajectory(levelName: string): Trajectory[] {
-    const resp = trajectories[levelName];
-    return resp ? resp.values : [];
-  }
-
-  function getConflicts(levelName: string): Conflict[] {
-    const resp = trajectories[levelName];
-    return resp ? resp.conflicts : [];
-  }
-
   const sortedMapFloorLayers = mapFloorLayerSort.map(x => mapFloorLayers[x]);
   const ref = React.useRef<ImageOverlay>(null);
 
   if (ref.current) {
     ref.current.leafletElement.setZIndex(0);
   }
 
+  React.useEffect(() => {
+    function getTrajectory(levelName: string): Trajectory[] {
+      const resp = trajectories[levelName];
+      return resp ? resp.values : [];
+    }
+
+    function getConflicts(levelName: string): Conflict[] {
+      const resp = trajectories[levelName];
+      return resp ? resp.conflicts : [];
+    }
+
+    function getConflictRobotsName(conflicts: Conflict[], trajs: Trajectory[]): string[][] {
+      let conflictRobotNames: string[][] = [];
+      if (conflicts.length !== 0) {
+        conflicts.forEach(conflictPair => {
+          let robotNames: string[] = [];
+          conflictPair.forEach(conflictId => {
+            const robotName = DefaultTrajectoryManager.getRobotNameFromPathId(conflictId, trajs);
+            robotName && robotNames.push(robotName);
+          });
+          robotNames && conflictRobotNames.push(robotNames);
+        });
+      }
+      return conflictRobotNames;
+    }
```

Comment:
What do you think if we do this? It looks less convoluted, right?

``` typescript
      if (conflicts.length === 0) {
        return conflictRobotNames;
      }

      conflicts.forEach(conflictPair => {
        let robotNames: string[] = [];
        conflictPair.forEach(conflictId => {
          const robotName = DefaultTrajectoryManager.getRobotNameFromPathId(conflictId, trajs);
          robotName && robotNames.push(robotName);
        });
        robotNames && conflictRobotNames.push(robotNames);
      });
```

Reply:
```
      if (conflicts.length !== 0) {
        return conflictRobotNames;
      }

      conflicts.forEach(conflictPair => {
        let robotNames: string[] = [];
        conflictPair.forEach(conflictId => {
          const robotName = DefaultTrajectoryManager.getRobotNameFromPathId(conflictId, trajs);
          robotName && robotNames.push(robotName);
        });
        robotNames && conflictRobotNames.push(robotNames);
      });
      return conflictRobotNames;
```

if the repeating return line is alright, then yea we can go with separating them!

Reply:
Yeah, we can return an empty array of type string[][] to avoid confusion on the first `conflictRobotNames`.